netman/adapters/switches/juniper/mx.py

```diff
@@ -85,7 +86,21 @@ def add_ip_to_vlan(self, vlan_number, ip_network):
         self._push(update)
 
     def remove_ip_from_vlan(self, vlan_number, ip_network):
-        raise NotImplementedError()
+        config = self.query(one_interface_vlan(vlan_number))
+
+        if len(config.xpath("data/configuration/interfaces/interface/unit")) < 1:
+            raise UnknownVlan(vlan_number)
+
+        for addr_node in config.xpath("data/configuration/interfaces/interface/unit/family/inet/address/name"):
+            address = IPNetwork(addr_node.text)
+            if ip_network in address:
+                update = Update()
+                update.add_interface(ip_network_element(vlan_number, ip_network, ' operation="delete"'))
+
+                self._push(update)
+                return
```

Comment:
IMO this is not readable, returning to prevent from raising an exception.
I would prefer to reverse the condition

```
    def remove_ip_from_vlan(self, vlan_number, ip_network):
        config = self.query(one_interface_vlan(vlan_number))

        if len(config.xpath("data/configuration/interfaces/interface/unit")) < 1:
            raise UnknownVlan(vlan_number)

        if not self._ip_in_unit_addresses(ip_network, config):
            raise UnknownIP(ip_network)

        update = Update()
        update.add_interface(ip_network_element(vlan_number, ip_network, ' operation="delete"'))

        self._push(update)

    def _ip_in_unit_addresses(self, ip_network, config):
        for addr_node in config.xpath("data/configuration/interfaces/interface/unit/family/inet/address/name"):
            address = IPNetwork(addr_node.text)
            if ip_network in address:
                return True
```